src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/ObjectExplorerService.cs

```diff
@@ -453,7 +453,24 @@ internal ExpandResponse QueueExpandNodeRequest(ObjectExplorerSession session, st
                                else
                                {
                                    Logger.Verbose($"Expanding {nodePath}");
-                                   nodes = node.Expand(cancelToken, securityToken?.Token, filters).Select(x => x.ToNodeInfo()).ToArray();
+                                   try
+                                   {
+                                       nodes = node.Expand(cancelToken, securityToken?.Token, filters).Select(x => x.ToNodeInfo()).ToArray();
+                                   }
+                                   catch (ConnectionFailureException ex)
+                                   {
+                                       Logger.Error($"Failed to expand node: {ex.InnerException.Message}");
+                                       var errorNode = new NodeInfo()

```

Comment:
I would suggest making a helper function for creating an error node to avoid copy/paste errors if we add more in the future. Put it as a static method on the NodeInfo class.